### PR TITLE
Eng 12597 recover from automatic snapshot when c/l is disabled

### PR DESCRIPTION
**Pull request by luyangco**

### CODE REVIEW

File: tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java [hunk 1]
```diff
@@ -3314,73 +3334,500 @@ else if (partitionId != 0) {
...
+    {
+        if (isValgrind()) return; // snapshot doesn't run in valgrind ENG-4034
+
+        System.out.println("Starting testMajorityOfClusterMissingSnapshot");
+        m_config.shutDown();
+
+        int num_replicated_items = 1000;
+        int num_partitioned_items = 126;
+
+        // 2 node k=1 cluster
```
Comment (akhanzode) [inline tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java]: 3 node k 1 right?

File: tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java [hunk 2]
```diff
@@ -3314,73 +3334,500 @@ else if (partitionId != 0) {
...
+                loadTable(client, "PARTITION_TESTER", false, another_partition_table);
+                // now reptable has 2000 rows and parttable has 252 rows
+                saveTablesWithPath(client, TESTNONCE + "2", snapshotsPath);
+            } finally {
+                client.close();
+            }
+
+            lc.shutDown();
+            boolean clearLocalDataDirectories = false;
+            lc.startUp(clearLocalDataDirectories);
```
Comment (akhanzode) [inline tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java]: why not just pass false?

File: tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java [hunk 3]
```diff
@@ -3314,73 +3334,500 @@ else if (partitionId != 0) {
...
+                long reptableRows = client.callProcedure("@AdHoc", "select count(*) from REPLICATED_TESTER").getResults()[0].asScalarLong();
+                assertEquals(2000, reptableRows);
+                long parttableRows = client.callProcedure("@AdHoc", "select count(*) from PARTITION_TESTER").getResults()[0].asScalarLong();
+                assertEquals(252, parttableRows);
+            } finally {
+                client.close();
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
```
Comment (akhanzode) [inline tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java]: Can you recover again just to make sure that terminus snapshot stays as the latest snapshot and is not deleted.

File: tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java (hunk 2 above)
Comment (luyangco) [reply tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java]: For readability?

File: tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java (hunk 1 above)
Comment (luyangco) [reply tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java]: No, in this test it's 2nk1.

File: tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java (hunk 3 above)
Comment (luyangco) [reply tests/frontend/org/voltdb/regressionsuites/TestSaveRestoreSysprocSuite.java]: Sure will do.

**Pull request merged**